internal/pkg/term/cursor/cursor.go

```diff
@@ -13,23 +14,54 @@ import (
 type cursor interface {
 	Up(n int)
 	Down(n int)
+	Hide()
+	Show()
+}
+
+// fakeFileWriter is a terminal.FileWriter.
+// If the underlying writer w does not implement Fd() then a dummy value is returned.
+type fakeFileWriter struct {
+	w io.Writer
+}
+
+// Write delegates to the internal writer.
+func (w *fakeFileWriter) Write(p []byte) (int, error) {
+	return w.w.Write(p)
+}
+
+// Fd is required to be implemented to satisfy the terminal.FileWriter interface.
+// If the underlying writer is a file, like os.Stdout, then invoke it. Otherwise, this method allows us to create
+// a Cursor that can write to any io.Writer like a bytes.Buffer by returning a dummy value.
+func (w *fakeFileWriter) Fd() uintptr {
+	if v, ok := w.w.(terminal.FileWriter); ok {
+		return v.Fd()
+	}
+	return 0
 }
 
 // Cursor represents the terminal's cursor.
 type Cursor struct {
 	c cursor
 }
 
-// New creates a new cursor that writes to stderr and reads from stdin.
+// New creates a new cursor that writes to stderr.
 func New() *Cursor {
 	return &Cursor{
 		c: &terminal.Cursor{
-			In:  os.Stdin,
```

Comment:
Why do we remove this one? Is it not used?

Reply:
It's not used yeah 😅